homeassistant/components/rachio/calendar.py

```diff
@@ -0,0 +1,180 @@
+            start_time = dt_util.parse_datetime(
+                self._previous_event[KEY_START_TIME], raise_on_error=True
+            )
+            end_time = start_time + timedelta(
+                seconds=int(self._previous_event[KEY_TOTAL_RUN_DURATION])
+            )
+            if start_time <= dt_util.now() <= end_time:
+                return self._previous_event
+
+        schedule = iter(self.coordinator.data)
+        event = next(schedule, None)
+        if not event:  # Schedule is empty
+            return None
+        while (
+            not event[KEY_SKIPPABLE] or KEY_SKIP in event[KEY_RUN_SUMMARIES][0]
+        ):  # Not being skippable indicates the event is in the past
+            event = next(schedule, None)
+            if not event:  # Schedule only has past or skipped events
+                return None
+        self._previous_event = event  # Store for future use
+        return event
+
+    async def async_get_events(
+        self, hass: HomeAssistant, start_date: datetime, end_date: datetime
+    ) -> list[CalendarEvent]:
+        """Get all events in a specific time frame."""
+        if not self.coordinator.data:
+            raise HomeAssistantError("No events scheduled")
+        schedule = self.coordinator.data
+        event_list: list[CalendarEvent] = []
+
+        for run in schedule:
+            event_start = dt_util.as_local(
+                dt_util.parse_datetime(run[KEY_START_TIME], raise_on_error=True)
+            )
+            if event_start > end_date:
+                break
+            if run[KEY_SKIPPABLE]:  # Future events
+                event_end = event_start + timedelta(
+                    seconds=int(run[KEY_TOTAL_RUN_DURATION])
+                )
+            else:  # Past events
+                event_end = event_start + timedelta(
+                    seconds=int(run[KEY_RUN_SUMMARIES][0][KEY_DURATION_SECONDS])
+                )
+
+            if (
+                event_end > start_date
+                and event_start < end_date
+                and KEY_SKIP not in run[KEY_RUN_SUMMARIES][0]
+            ):
+                valves = ", ".join(
+                    [event[KEY_VALVE_NAME] for event in run[KEY_RUN_SUMMARIES]]
+                )
+                event = CalendarEvent(
+                    summary=run[KEY_PROGRAM_NAME],
+                    start=event_start,
+                    end=event_end,
+                    description=valves,
+                    location=self._location,
+                    uid=f"{run[KEY_PROGRAM_ID]}/{run[KEY_START_TIME]}",
+                )
+                event_list.append(event)
+        return event_list
+
+    async def async_delete_event(
+        self,
+        uid: str,
+        recurrence_id: str | None = None,
+        recurrence_range: str | None = None,
+    ) -> None:
+        """Skip an upcoming event on the calendar."""
+        event = uid.split("/")
+        program = event[0]
+        timestamp = event[1]
```

Comment:
```suggestion
        program, timestamp = uid.split("/")
```

If its always one `/`